avocado/utils/archive.py

```diff
@@ -165,26 +165,31 @@ def extract(self, path='.'):
         """
         self._engine.extractall(path)
         if self.is_zip:
-            self._update_symlinks(path)
+            self._update_zip_extra_attrs(path)
 
-    def _update_symlinks(self, dst_dir='.'):
+    def _update_zip_extra_attrs(self, dst_dir):
         if platform.system() != "Linux":
-            LOG.warn("Symlink handling in zip files not supported on Windows.")
+            LOG.warn("Attr handling in zip files only supported on Linux.")
             return
         # Walk all files and re-create files as symlinks
         for path, info in self._engine.NameToInfo.iteritems():
-            if info.external_attr >> 16 & stat.S_IFLNK == stat.S_IFLNK:
+            dst = os.path.join(dst_dir, path)
+            if not os.path.exists(dst):
+                msg = ("Paths in this zip file are stored in unsupported "
+                       "format, not updating the attributes. (%s)"
+                       % path)
+                LOG.warn(msg)
```

Comment:
There's only a slight style change I'd like to have to merge it, which is using `LOG.warn("...", path)`.  I'm adding that to this commit and will merge it.